app/models/alchemy/page/page_naming.rb

```diff
@@ -56,6 +55,17 @@ def external_urlname
       "http://#{urlname}"
     end
 
+    # Returns an array of visible/non-language_root ancestors.
+    def visible_ancestors
+      if parent && new_record?
```

Comment:
It would be great to have an earlier if exit: `if new_record? && parent`, so then only when the record is `new` the check for `parent` happens. Also that's more readable and much clearer why this recursion need to happen.